spec/services/reblog_service_spec.rb

```diff
@@ -37,11 +37,22 @@
   context 'when the reblogged status is discarded in the meantime' do
     let(:status) { Fabricate(:status, account: alice, visibility: :public) }
 
+    # Add a callback to discard the status being reblogged after the
+    # validations pass but before the database commit is executed.
     before do
-      # Update the in-database attribute without reflecting the change in
-      # the object. This cannot simulate all race conditions, but it is
-      # pretty close.
-      Status.where(id: status.id).update_all(deleted_at: Time.now.utc) # rubocop:disable Rails/SkipsModelValidations
+      Status.class_eval do
+        before_save :discard_status
+        def discard_status
+          Status
+            .where(id: reblog_of_id)
+            .update_all(deleted_at: Time.now.utc) # rubocop:disable Rails/SkipsModelValidations
+        end
+      end
+    end
+
+    # Remove race condition simulating `discard_status` callback.
+    after do
+      Status._validation_callbacks.delete(:discard_status)
```

Comment:
This needs to be changed now that the callback has changed:
```suggestion
      Status._save_callbacks.delete(:discard_status)
```